lib/modules/manager/asdf/extract.spec.ts

```diff
@@ -344,6 +345,13 @@ dummy 1.2.3
             depName: 'lua',
             extractVersion: '^v(?<version>\\S+)',
           },
+          {
+            currentValue: '3.9.5',
+            datasource: 'github-releases',
```

Comment:
use maven datasource, see maven wrapper manager for sample

Reply:
```suggestion
            datasource: 'maven',
```

Reply:
I'll try running some tests, but I've drafted suggestions now on what I think you may be after.
Let me know if you see something else I haven't already.